src/vivarium_cluster_tools/psimulate/paths.py

```diff
@@ -70,14 +72,29 @@ def from_entry_point_args(
         cls,
         *,
         command: str,
-        input_artifact_path: Optional[Path],
         result_directory: Path,
+        input_artifact_path: Optional[Path] = None,
+        input_model_spec_path: Path,
     ) -> "OutputPaths":
         launch_time = datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
 
         output_directory = result_directory
         if command == COMMANDS.run:
-            output_directory = output_directory / input_artifact_path.stem / launch_time
+            # Peek at the CLI args and model spec YAML to determine a good subdirectory name
+            # for the results. If we don't have an artifact, use the stem of the model spec,
+            # which must exist. NB: Actual determination of the artifact path is done in the
+            # model_specification.py code.
+            if input_artifact_path:
+                location = input_artifact_path.stem
+            else:
+                # check for YAML
+                with open(input_model_spec_path) as model_spec_file:
+                    model_spec = yaml.safe_load(model_spec_file)
+                try:
+                    location = Path(model_spec["configuration"]["input_data"]["artifact_path"]).stem
```

Comment:
In what context would `input_artifact_path` not exist, but there be an artifact path defined in the model spec file? This seems like it should never happen.

Reply:
None, but you still have to look. Three possibilities:

CLI
YAML
none

In the first two we use the artifact name, in the latter we use the name of the YAML file.

Reply:
Oh, you mean why specify artifact in the YAML but not the CLI? 

I prefer it tbqh. Fewer options to pass to psimulate.